dgraphtest/local_cluster.go

```diff
@@ -623,7 +623,15 @@ func (c *LocalCluster) HTTPClient() (*HTTPClient, error) {
 	if err != nil {
 		return nil, err
 	}
-	return &HTTPClient{adminURL: adminURL, graphqlURL: graphqlURL}, nil
+	licenseURL, err := c.licenseURL()
+	if err != nil {
+		return nil, err
+	}
+	stateURL, err := c.stateURL()
+	if err != nil {
+		return nil, err
+	}
+	return &HTTPClient{adminURL: adminURL, graphqlURL: graphqlURL, licenseURL: licenseURL, stateURL: stateURL}, nil
```

Comment:
I think now we have too many URLs. I am thinking, instead of having this function in the concrete types (LocalCluster, ComposeCluster), we can instead define functions for baseURLs and then define these functions directly cluster.go. This will avoid repetition of the code of appending the constant part of the URL to the base part. What do you think?

Reply:
Yes, make sense, I have made changes, please check.